cli/config_resolver.go

```diff
@@ -0,0 +1,57 @@
+package cli
```

Comment:
I've already noticed that most of the files in the `cli` package are missing the Apache 2 license header. This shouldn't be addressed in this PR but rather tackled once https://github.com/prometheus/prometheus/issues/3904 is fixed.